airflow/dags/ingestion/ask-astro-load-cosmos-docs.py

```diff
@@ -0,0 +1,59 @@
+import os
+from datetime import datetime
+
+from include.utils.slack import send_failure_notification
+
+from airflow.decorators import dag, task
+from airflow.providers.weaviate.operators.weaviate import WeaviateDocumentIngestOperator
+
+ask_astro_env = os.environ.get("ASK_ASTRO_ENV", "dev")
+
+_WEAVIATE_CONN_ID = f"weaviate_{ask_astro_env}"
+WEAVIATE_CLASS = os.environ.get("WEAVIATE_CLASS", "DocsDev")
+
+
+default_args = {"retries": 3, "retry_delay": 30}
+
+schedule_interval = os.environ.get("INGESTION_SCHEDULE", "0 5 * * 2") if ask_astro_env == "prod" else None
+
+
+@task
+def get_cosmos_docs_content():
+    from include.tasks.extract.cosmos_docs import extract_cosmos_docs
+
+    return extract_cosmos_docs()
+
+
+@dag(
+    schedule_interval=schedule_interval,
+    start_date=datetime(2023, 9, 27),
+    catchup=False,
+    is_paused_upon_creation=True,
+    default_args=default_args,
+    on_failure_callback=send_failure_notification(
+        dag_id="{{ dag.dag_id }}", execution_date="{{ dag_run.execution_date }}"
+    ),
+)
+def ask_astro_load_cosmos_docs():
+    """
+    This DAG performs incremental load for any new Cosmos docs. Initial load via ask_astro_load_bulk imported
+    data from a point-in-time data capture. By using the upsert logic of the weaviate_import decorator
+    any existing documents that have been updated will be removed and re-added.
+    """
+
+    from include.tasks import split
+
+    split_docs = task(split.split_html).expand(dfs=[get_cosmos_docs_content()])
```

Comment:
Presumably not in scope for this PR, but why wouldn't the `split.split_html` "task" already be decorated with `@task`? Is the idea that the function could be used as a standalone function and not necessarily as a task within a DAG?

Reply:
I agree. Similar to the returning list of DF comment, I was also thinking about the same thing but wasn't sure what the previous team/owner had in consideration. I can make a new issue to tackle this in a separate PR for refactoring as a refactor here may make this PR much harder to review.